### PR TITLE
feat: add --from/--to date-range filtering for daily and monthly views

### LINKED ISSUE #223: Add --from/--to to daily and monthly view for more fine granular filtering
## Summary

`--view daily` and `--view monthly` print every period in the local logs. There's no way to ask for just a window. If you want June out of six months of data, you scroll past everything else and add it up by hand.

## Proposal

`--from` and `--to` flags that trim the table to an inclusive range. Daily takes `YYYY-MM-DD`, monthly takes `YYYY-MM`, matching what each view already prints. Allowing just `--from` or just `--to` covers the "everything since X" case.

### PR DESCRIPTION
**Pull request by TobiWo**
## Summary

Adds `--from`/`--to` to restrict the `daily` and `monthly` tables to an inclusive date range.

Format follows the view: `daily` takes `YYYY-MM-DD`, `monthly` takes `YYYY-MM`. Either bound works alone. The flags only apply to those two views — using them elsewhere is a validation error.

Most of the plumbing already existed. `UsageAggregator` accepts `start_date`/`end_date` and filters on the entry timestamp; the public `aggregate()` just never passed them. So this connects the flags to that filter instead of writing new aggregation. The totals row and period header follow automatically, since both come from the rows the aggregator returns.

One thing reviewers should know: these views read `~/.claude/projects/`, which Claude Code prunes after `cleanupPeriodDays` (default 30). A range older than that returns no rows, not an error. The README explains how to raise the retention for longer monthly history.

Changes: the validated `--from`/`--to` fields on `Settings`, a `_range_bounds()` helper that builds timezone-aware inclusive bound (month-end via `calendar.monthrange`), tests for both views including boundary inclusivity and the error cases, and the README entries.

Closes #223 

## Summary by CodeRabbit

* **New Features**
  * Added optional `--from` / `--to` date-range filters for daily and monthly table views (aliases supported).
  * Filtering is inclusive and applied during aggregation; supports one-sided ranges.
* **Bug Fixes**
  * Enforces correct ISO formats per view, validates calendar dates, and rejects invalid/ordered ranges and unsupported views.
  * Bounds are interpreted using UTC-based daily keys even when display timezone is non-UTC; date-range replay now respects saved view vs explicitly provided view.
* **Documentation**
  * Updated README with a “Filtering daily/monthly views by date range” section, examples, and retention notes.
* **Tests**
  * Added coverage for inclusive bounds, edge cases (including leap years), and timezone/key behavior.

### CODE REVIEW

File: src/claude_monitor/core/settings.py [hunk 1]
```diff
@@ -273,6 +274,22 @@ def _get_system_time_format() -> str:
...
+    )
+
+    date_to: Optional[str] = Field(
+        default=None,
+        alias="to",
+        description=(
+            "End of date range, inclusive. daily: YYYY-MM-DD, monthly: YYYY-MM"
+        ),
+    )
+
```
[the remark this answers is not in the document]
Comment (TobiWo) [reply src/claude_monitor/core/settings.py:292]: `date_from`/`date_to` are never written to last-used (excluded from LastUsedParams.save), so the stale-overwrite can't occur today. Applied the alias -> field mapping anyway as defensive hygiene for any future aliased field.

**Comment by TobiWo** [comment]
@Maciek-roboblog All comments raised by coderabbit have been adressed.
